src/Bridge/Doctrine/MongoDbOdm/Filter/SearchFilter.php

```diff
@@ -166,20 +166,24 @@ private function addEqualityMatchStrategy(string $strategy, string $field, $valu
     {
         $type = $metadata->getTypeOfField($field);
 
+        if (MongoDbType::STRING !== $type) {
+            return MongoDbType::getType($type)->convertToDatabaseValue($value);
+        }
+
+        $quotedValue = preg_quote($value);
```

Comment:
Shouldn't we use the doctrine quote method? https://www.doctrine-project.org/projects/doctrine-dbal/en/2.12/reference/security.html#right-quoting-escaping-values

Reply:
Or maybe we should use parameters. We shouldn't have to escape the values, I think there are security issues here.

Reply:
We're not escaping SQL here. We're escaping regular expression characters. Thus I don't think the quoting link applies, or that parameters can be used in this case.

Ideally, the `Regex` class would have a static method to escape the necessary characters, but it doesn't seem to have a method like that.

Here's a [relevant Stack Overflow question](https://stackoverflow.com/questions/27962072/how-to-escape-special-characters-using-mongo-regex-in-php-format) from 5 years ago.